packages/YamlRector/src/Rector/RenameSubKeyYamlRector.php

```diff
@@ -57,35 +34,59 @@ public function isCandidate(string $content): bool
     }
 
     public function refactor(string $content): string
+    {
+        foreach ($this->pathsToNewKeys as $path => $newKey) {
+            $pathPattern = $this->createPatternFromPath($path);
+
+            while (Strings::match($content, $pathPattern)) {
+                $replacement = $this->createReplacementFromPathAndNewKey($path, $newKey);
+                $content = Strings::replace($content, $pathPattern, $replacement, 1);
+            }
+        }
+
+        return $content;
+    }
+
+    private function createPatternFromPath(string $path): string
+    {
+        $pathParts = $this->splitPathToParts($path);
+
+        $pattern = '';
+        foreach ($pathParts as $key => $pathPart) {
+            if ($key === (count($pathParts) - 1)) {
+                // last only up-to the key name
+                $pattern .= sprintf('(%s)(.*?)', preg_quote($pathPart));
+            } else {
+                // see https://regex101.com/r/n8XPbV/3/ for ([^:]*?)
+                $pattern .= sprintf('(%s:\s+)([^:]*?)', preg_quote($pathPart));
```

Comment:
Idea: Instead of all,match newline + at least 4 spaces (per nesting) then anything